tests/tests_unit/test_data_classes/test_data_models/test_views.py

```diff
@@ -1,33 +1,47 @@
-import pytest
-
-from cognite.client.data_classes.data_modeling.views import ViewPropertyDefinition
+from cognite.client.data_classes.data_modeling.views import ViewProperty
 
 
 class TestViewPropertyDefinition:
-    @pytest.mark.parametrize(
-        "data",
-        [
-            {
-                "type": {"type": "text", "list": False, "collation": "ucs_basic"},
-                "container": {"space": "mySpace", "externalId": "myExternalId", "type": "container"},
-                "containerPropertyIdentifier": "name",
-                "description": None,
-                "name": "fullName",
-                "nullable": False,
-                "autoIncrement": False,
-                "defaultValue": None,
-            },
-            {
-                "type": {"space": "mySpace", "externalId": "myExternalId"},
-                "source": {"type": "view", "space": "mySpace", "externalId": "myExternalId", "version": "myVersion"},
-                "direction": "outwards",
-                "description": None,
-                "edgeSource": None,
-                "name": "fullName",
-            },
-        ],
-    )
-    def test_load_dump(self, data: dict):
-        actual = ViewPropertyDefinition.load(data).dump(camel_case=True)
+    def test_load_dumped_mapped_property(self) -> None:
+        input = {
+            "type": {"type": "text", "list": False, "collation": "ucs_basic"},
+            "container": {"space": "mySpace", "externalId": "myExternalId", "type": "container"},
+            "containerPropertyIdentifier": "name",
+            "description": None,
+            "name": "fullName",
+            "nullable": False,
+            "autoIncrement": False,
+            "defaultValue": None,
+        }
+        actual = ViewProperty.load(input)
+
+        assert actual.dump() == {
+            "auto_increment": False,
+            "container": {"external_id": "myExternalId", "space": "mySpace"},
```

Comment:
Why is the `"type": container"` no longer in the dump? Should .load().dump() result in the same data?

Reply:
I see now that `ViewProperty` is an abstract class and does do not produce the same subclass.